hngx-org/authentication_service
# Microservices authentication api now working

## Description
Implementation of authentication and authorisation gate way for communications with other microservices
​
## Related Issue (Link to linear ticket)
https://linear.app/zuri-project-backend/issue/AUT-21/creation-of-endpoints-for-adding-and-revoking-user-permissions
​
## Motivation and Context
This change is necessary inorder to link the authentication microservice with all other microservices.
this will help them to ensure that any user visiting is authorised or authenticated
​
## How Has This Been Tested?
This has been tested on a locally hosted database, with edits made to the models to ensure preoper relationship between user and role, roles and permissions, users and permissions
​
1. Ensure postman is installed.
2. Make a post request 
Download, Install and run postman


make a post request containing jwt obtained from a user request header
with either role or permission to validate in the post body

Test images in linear app link


POST   https://auth.akuya.tech/api/get-auth
```

{

"token": "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6ImUwNzY0MmQ1LWU0NDUtNGQ3OS1hMmI0LWUzMDc4ZGQwZDBiMiIsImlhdCI6MTY5Njg3MTE5Nn0.r5-H0Xp-SOJT2KdDFBREufbLuRQCTbb9KxkLuCKpK7U"

}
```
valid response

```
{

"status": 200,

"id": "e07642d5-e445-4d79-a2b4-e3078dd0d0b2"

}

```
POST   https://auth.akuya.tech/api/get-auth
```
{

"token": "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6ImUwNzY0MmQ1LWU0NDUtNGQ3OS1hMmI0LWUzMDc4ZGQwZDBiMiIsImlhdCI6MTY5Njg3MTE5Nn0.r5-H0Xp-SOJT2KdDFBREufbLuRQCTbb9KxkLuCKpK7Y"

}
```
Not Authenticated and not authorised response

```
{

"status": 401,

"msg": "Unauthorized"

}
```
Test with role included
```
{

"token": "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6ImUwNzY0MmQ1LWU0NDUtNGQ3OS1hMmI0LWUzMDc4ZGQwZDBiMiIsImlhdCI6MTY5Njg3MTE5Nn0.r5-H0Xp-SOJT2KdDFBREufbLuRQCTbb9KxkLuCKpK7U",

"role": "Registered User"

}
```
Test with permission
```
{

"token": "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6ImUwNzY0MmQ1LWU0NDUtNGQ3OS1hMmI0LWUzMDc4ZGQwZDBiMiIsImlhdCI6MTY5Njg3MTE5Nn0.r5-H0Xp-SOJT2KdDFBREufbLuRQCTbb9KxkLuCKpK7U",

"permission": "create product"

}
```
Response if authorised
```

{

"status": 200,

"msg": "authorized",

"id": "e07642d5-e445-4d79-a2b4-e3078dd0d0b2"

}
```

here is a link to available permissions and roles to test for
https://auth.akuya.tech/api/get-auth/roles-and-permissions

Test permissions route
POST https://auth.akuya.tech/api/get-auth/permissions
```

{

"token": "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6ImUwNzY0MmQ1LWU0NDUtNGQ3OS1hMmI0LWUzMDc4ZGQwZDBiMiIsImlhdCI6MTY5Njg3MTE5Nn0.r5-H0Xp-SOJT2KdDFBREufbLuRQCTbb9KxkLuCKpK7U"

}
```
Response when validated
```
{

"status": 200,

"permissions": [

"create product",

"view product",

"edit product",

"delete own product",

"view own products",

"view all products",

"use coupon",

"access checkout",

"purchase product",

"view own orders",

"delete order",

"create sales report",

"view all shops",

"create portfolio",

"update own portfolio",

"view own portfolio",

"view all portfolios",

"delete own portfolio",

"take assessment",

"view own results",

"create event",

"delete own event",

"delete all events",

"edit event",

"edit all events",

"view events"

]

}
```
Response if not valid or expired token
```

{

"status": 401,

"msg": "Unauthorized"

}
```
## Screenshots (if appropriate - Postman, etc):
​
## Types of changes

​
## Checklist:


Collapse
